src/main/java/seedu/duke/ui/TextUi.java

```diff
@@ -0,0 +1,136 @@
+    private static final String INDENTATION_SPACES = "    ";
+    private static final int DEFAULT_SEPARATOR_LENGTH = 40;
+    private static final int EMPTY_MESSAGE = 0;
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private String separator = "-";
+
+    /**
+     * Print spaces before output message for better formatting.
+     */
+    public void printIndentation() {
+        System.out.print(INDENTATION_SPACES);
+    }
+
+    /**
+     * Read user's input command and return back the command string.<br/>
+     *
+     * @return User input command with leading/dangling whitespace being removed
+     */
+    public String getCommand() {
+        String userCommand = "";
+        try {
+            String inputLine = SCANNER.nextLine();
+            userCommand = inputLine.trim();
+        } catch (BufferOverflowException bufferOverFlowException) {
+            printErrorFor(bufferOverFlowException);
+        } catch (NoSuchElementException noElementException) {
+            printErrorFor(noElementException);
+        }
+        return userCommand;
+    }
+
+    /**
+     * Customise separators for each feature.<br/>
+     * <br/>
+     * At this moment we can only use length == 1 separator for consistency of length of line separator.<br/>
+     * This will be improved on in the future to allow for more patterns.
+     *
+     * @param separator Length == 1 string
+     */
+    public void setSeparator(String separator) {
+        this.separator = separator;
+    }
+
+    /**
+     * Print line separators for output lines.<br/>
+     * <br/>
+     * Each subclass inherited from this class can override this method to vary the interface.
+     */
+    public void printSeparator() {
+        for (int i = 0; i < DEFAULT_SEPARATOR_LENGTH; i += 1) {
+            System.out.print(separator);
+        }
+        System.out.print(System.lineSeparator());
+    }
+
+    /**
+     * Split a message string to a string array using System.lineSeparator().<br/>
+     * <br/>
+     * This is to help indent each new line during output.<br/>
+     * Each line of string will be trimmed to remove leading/dangling whitespace.
+     *
+     * @param message Message to be printed
+     * @return Messages that are split using lineSeparator
+     */
+    private String[] splitOutputMessage(String message) {
+        String[] newLineMessages = message.trim().split(System.lineSeparator());
+        for (int i = 0; i < newLineMessages.length; i += 1) {
+            newLineMessages[i] = newLineMessages[i].trim();
+        }
+        return newLineMessages;
+    }
+
+    /**
+     * Print each new line of message on a separate line with indentation being added.<br/>
+     * <br/>
+     * Output message with one line is also accepted.
+     *
+     * @param message Output message to be printed
+     */
+    private void printMultiLineMessage(String message) {
+        String[] newLineMessage = splitOutputMessage(message);
```

Comment:
Perhaps rename `newLineMessage` to `newLineMessages`? Since this is an array of `String`